onnxruntime/contrib_ops/cuda/diffusion/bias_add_impl.cu

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// The CUDA kernel is modified from SeqLen2Spatial plugin of TensorRT 8.5.
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 1993-2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cub/cub.cuh>
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "contrib_ops/cuda/diffusion/bias_add_impl.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+template <typename T, int32_t C, int32_t TPB>
+__global__ void BiasAddKernel(T const* input, T const* bias, T const* residual, T* output) {
+  int32_t base_offset = blockIdx.x * C + threadIdx.x;
+  int32_t bias_offset = threadIdx.x;
+
+#pragma unroll
+  for (int32_t i = 0; i < C / TPB; ++i) {
+#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 530)
+    output[base_offset] = input[base_offset] + bias[bias_offset] + residual[base_offset];
+#else
+    output[base_offset] = static_cast<T>(float(input[base_offset]) + float(bias[bias_offset]) + float(residual[base_offset]));
+#endif
```

Comment:
include cu_inc/common.cuh?

Reply:
Included (see above) but the amd build still failed so I have to add this.

Reply:
oh, i see. you need to add 'using namespace onnxruntime::cuda;' to make compiler know the override operators